Shared/Services/NftMetadataService.cs

```diff
@@ -60,7 +61,11 @@ public void Dispose()
             {
                 request.Timeout = 10000; //we can't afford to wait forever here, 10s must be enough
                 var response = await _client.GetAsync(request, cancellationToken);
-                return JsonConvert.DeserializeObject<NftMetadata>(response.Content!);
+                var token = JToken.Parse(response.Content!);
+                var json = token.ToObject<NftMetadata>();
+                if ((token != null) && (json != null))
+                    json.JSONContent = token.ToString(Formatting.Indented);
+                return json;
```

Comment:
Could do this:
```C#
var metaData = JsonConvert.DeserializeObject<NftMetadata>(response.Content!);
metaData!.JSONContent = metaData != null 
    ? JsonConvert.SerializeObject(metaData, Formatting.Indented) 
    : "";
return metaData;
```

Reply:
I explicitly did *not* want to do this - what's the point of doing the round trip with serialising again what we just deserialised? I want the original JSON response, not something we've generated.

Reply:
Oh but you're definitely right about the var names, I've changed `json` to `metadata` now and it reads a lot better.

Reply:
I can see that if the metadata holds properties we don't have in the class, I don't know if this is the case often.  
Round trip with serializing would only be for the purpose of formatting the string. 

It's the same with parsing the response as JToken, doing a round trip for formatting purpose, the difference is ofc as you say it will take the original response no matter what the string is. 👍🏻

Reply:
There are people that want to include extra metadata properties that also want to display so modersohn has a good approach to this. I'm closing this out